### PR TITLE
Memcached Support

### PR DESCRIPTION
**Pull request by MatinRh**
Issue #18 is done.

I really appreciate it if you label this PR as ```hacktoberfest-accepted```.

### CODE REVIEW

File: MemcachedChecker/MemcachedChecker.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	m.Host = cx.Host
+	m.Port = cx.Port
+}
+
+func (m *MemcachedConnection) Validate() (bool, error) {
+	if len(m.Host) == 0 {
+		return false, errors.New("Host can't be empty")
+	}
+
+	if m.Port < 0 || m.Port > 65535 {
```
Comment (ph4r5h4d) [inline MemcachedChecker/MemcachedChecker.go:20]: `m.Port < 1` is the correct way, kindly update it.

**Review by ph4r5h4d — CHANGES_REQUESTED** [review]
Kindly update the changes I've requested, I'll merge and release this Memcached module after RabbitMQ update. If any conflict happens, I'll resolve it.

Thanks for the time you spent on this, I really appreciate it.

File: MemcachedChecker/MemcachedChecker.go (hunk 1 above)
Comment (MatinRh) [reply MemcachedChecker/MemcachedChecker.go]: I've been confused. You did this port checking in other checkers with different approaches and hence I just decided to choose one of them.

File: MemcachedChecker/MemcachedChecker.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+func (m *MemcachedConnection) Check() (bool, bool, error) {
+	mc := memcache.New(m.BuildConnectionString())
+
+	// Since the current version of the Memcached Golang client doesn't create
+	// any connection to the Memcached server (or at least the client that I'm using)
+	// at the New method and at here we don't have any way for checking the correctness
+	// of the given server address, we must use another way for checking connection. As I
+	// understand, it seems the connection will establish at setting and getting items time,
+	// and I think this is a good point for checking connection.
+	err := mc.Set(&memcache.Item{Key: "foo", Value: []byte("bar")})
```
[the remark this answers is not in the document]
Comment (MatinRh) [reply MemcachedChecker/MemcachedChecker.go]: Sorry. I didn't read the entire documentation and hence I didn't notice of the existence of the Ping method.

**Comment by MatinRh** [comment]
The change request is done. Also, I resolved the merge conflicts.

**Comment by ph4r5h4d** [comment]
Thank you, I need to make an update to your PR because in another PR a slight change on one interface happened. thank you. will merge it today.

**Pull request merged**